pavelib/prereqs.py

```diff
@@ -195,22 +195,12 @@ def uninstall_python_packages():
     # to really really get rid of it.
     for _ in range(3):
         uninstalled = False
-        frozen = sh("pip freeze", capture=True).splitlines()
-
-        # Uninstall South
-        if any(line.startswith("South") for line in frozen):
-            sh("pip uninstall --disable-pip-version-check -y South")
-            uninstalled = True
-
-        # Uninstall edx-val
-        if any("edxval" in line for line in frozen):
-            sh("pip uninstall --disable-pip-version-check -y edxval")
-            uninstalled = True
-
-        # Uninstall django-storages
-        if any("django-storages==" in line for line in frozen):
-            sh("pip uninstall --disable-pip-version-check -y django-storages")
-            uninstalled = True
+        frozen = sh("pip freeze", capture=True)
+
+        for package in ("South", "edxval", "django-storages", "django-oauth2-provider"):
+            if "{}==".format(package) in frozen:
```

Comment:
This doesn't do the same thing as the original code, because edxval was installed with -e, and so wouldn't be found by this new code.  It's probably a moot point now, since people have had to make new devstacks for 1.8 anyway, but we should get a more robust technique here...